DEV/org.openl.core/src/org/openl/binding/impl/MethodNodeBinder.java

```diff
@@ -121,6 +124,55 @@ public IBoundNode bind(ISyntaxNode node, IBindingContext bindingContext) throws
         }
     }
 
+    private void validateMethodParameter(IMethodCaller methodCaller, IBoundNode[] children, ISyntaxNode node, IBindingContext bindingContext) {
+        IOpenClass[] parameterTypes = methodCaller.getMethod().getSignature().getParameterTypes();
+        if (parameterTypes.length == children.length) {
+            for (int i = 0; i < parameterTypes.length; i++) {
+                if (parameterTypes[i].getDomain() != null) {
+                    IDomain<Object> domain = (IDomain<Object>) parameterTypes[i].getDomain();
+                    String invalidValuesString = null;
+                    if (children[i] instanceof LiteralBoundNode) {
+                        LiteralBoundNode literalBoundNode = (LiteralBoundNode) children[i];
+                        if (literalBoundNode.getValue() != null && !domain.selectObject(literalBoundNode.getValue())) {
+                            invalidValuesString = literalBoundNode.getValue().toString();
+                        }
+                    } else if (children[i] instanceof ArrayInitializerNode) {
+                        // In case of MultiCallOpenMethod
+                        StringBuilder invalidValues = new StringBuilder();
+                        validateParameterArray(((ArrayInitializerNode) children[i]).children, domain, invalidValues);
+                        if (invalidValues.length() > 0) {
+                            invalidValuesString = invalidValues.substring(0, invalidValues.length() - 1);
+                        }
+                    }
+                    if (invalidValuesString != null) {
+                        BindHelper.processError(String.format(
+                                "Object '%s' is outside of valid domain '%s'. The comparison always returns %s.",
+                                invalidValuesString,
+                                DomainUtils.toString(domain),
+                                "ne".equals(methodCaller.getMethod().getName())), node, bindingContext);
```

Comment:
Remove `The comparison always returns false.` string. Where is the comparison in the arguments?